velox/functions/iceberg/BucketFunction.cpp

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/functions/iceberg/BucketFunction.h"
+#include "velox/common/base/Status.h"
+#include "velox/functions/Macros.h"
+#include "velox/functions/Registerer.h"
+#include "velox/functions/iceberg/util/Murmur3Hash32.h"
+#include "velox/type/Timestamp.h"
+
+namespace facebook::velox::functions::iceberg {
+namespace {
+
+FOLLY_ALWAYS_INLINE int apply(int numBuckets, int hashedValue) {
```

Comment:
Consider the name `bucketIndex` for a clearer definition.

Reply:
The `numBuckets` origins from Iceberg java code, returns the `bucketIndex`

Reply:
Can you think of another proper function name? The `apply` may be obscure.